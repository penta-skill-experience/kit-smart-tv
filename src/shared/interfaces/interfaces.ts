import {FontSize} from "../values/FontSize";
import {ColorScheme} from "../values/ColorScheme";
import {Announcement} from "../values/Announcement";

/**
 * This file describes the structure of the objects that are returned by and passed to the database.
 */

export interface ValuesData {
    fontSizes: FontSize[];
    colorSchemes: ColorScheme[];
}

export interface ConfigData {
    fontSize: string;
    colorScheme: string;
    background: string;
}

export interface WidgetDataData {
    widgetDataList: {
        widgetId: string;
        location: number;
        rawConfig?: Object;
    }[];
}

export interface AnnouncementsData {
    announcementDataList: Announcement[];
}