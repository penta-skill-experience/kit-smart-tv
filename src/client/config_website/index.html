<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>Admin Interface</title>
    <meta content="width=device-width, initial-scale=1" name="viewport">
    <link
            href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700&display=swap"
            rel="stylesheet"
    />
    <style>
        #root {
            height: 100vh;
            width: 100vw;
        }
    </style>
</head>
<body>
<div id="root">
    <!-- react output will go inside here -->
</div>
</body>
</html>
