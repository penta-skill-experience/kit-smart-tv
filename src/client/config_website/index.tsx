import "./index.css";
import * as ReactDOM from "react-dom";
import * as React from "react";
import {WidgetLoader} from "../widget/WidgetLoader";
import {ConcreteWidgetConfigSaver} from "./ConcreteWidgetConfigSaver";

ReactDOM.render(
    // this is just an example to demonstrate how widgets are instantiated
    // todo: remove this.
    <div>
        <h1>Hello Admin!</h1>
        <div>
            {new WidgetLoader().getWidget("rss-feed").createConfigComponent({url: "www.example.com"}, new ConcreteWidgetConfigSaver())}
        </div>
    </div>,
    document.getElementById("root")
);