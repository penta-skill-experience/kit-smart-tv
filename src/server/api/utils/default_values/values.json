{
  "fontSizes": [
    {
      "id": "small",
      "name": "small",
      "relativeSize": 0.8
    },
    {
      "id": "medium",
      "name": "medium",
      "relativeSize": 1.0
    },
    {
      "id": "large",
      "name": "large",
      "relativeSize": 1.15
    }
  ],
  "colorSchemes": [
    {
      "id": "dark",
      "name": "dark",
      "titleFontColor": "white",
      "bodyFontColor": "white",
      "specialBoldFontColor": "ForestGreen",
      "specialSubtleFontColor": "Tomato",
      "accentBarColor": "rgba(0, 0, 0, 0.5)",
      "backgrounds": [
        "https://images.pexels.com/photos/1809644/pexels-photo-1809644.jpeg",
        "https://images.pexels.com/photos/847402/pexels-photo-847402.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
        "https://images.pexels.com/photos/2162442/pexels-photo-2162442.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/4328298/pexels-photo-4328298.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/2085998/pexels-photo-2085998.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://wallpaperaccess.com/full/1379469.jpg",
        "https://images.pexels.com/photos/8824641/pexels-photo-8824641.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/9660579/pexels-photo-9660579.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
      ]
    },
    {
      "id": "light",
      "name": "light",
      "titleFontColor": "black",
      "bodyFontColor": "black",
      "specialBoldFontColor": "green",
      "specialSubtleFontColor": "FireBrick",
      "accentBarColor": "rgba(255,255,255,0.5)",
      "backgrounds": [
        "https://images.pexels.com/photos/8849653/pexels-photo-8849653.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/1323550/pexels-photo-1323550.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
        "https://images.pexels.com/photos/691668/pexels-photo-691668.jpeg",
        "https://images.pexels.com/photos/1367192/pexels-photo-1367192.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/4762392/pexels-photo-4762392.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/9716295/pexels-photo-9716295.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2",
        "https://images.pexels.com/photos/2101187/pexels-photo-2101187.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260",
        "https://images.pexels.com/photos/911738/pexels-photo-911738.jpeg?auto=compress&cs=tinysrgb&w=1260&h=750&dpr=2"
      ]
    }
  ]
}