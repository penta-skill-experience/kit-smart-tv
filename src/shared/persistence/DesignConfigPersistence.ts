import {ConfigData} from "./data";
import config from "./persistence.config.json";

export class DesignConfigPersistence {

    getSelectedColorSchemeId(): Promise<string> {
        return this.getConfigData().then(configData => configData.colorScheme);
    }

    getSelectedBackground(): Promise<string> {
        return this.getConfigData().then(configData => configData.background);
    }

    getSelectedFontSizeId(): Promise<string> {
        return this.getConfigData().then(configData => configData.fontSize);
    }

    setSelectedColorSchemeId(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.colorScheme = id;
                return this.setConfigData(configToSet)
            })
    }

    setSelectedBackground(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.background = id;
                return this.setConfigData(configToSet)
            });
    }

    setSelectedFontSize(id: string) {
        return this.getConfigData()
            .then(configData => {
                const configToSet = configData;
                configToSet.fontSize = id;
                return this.setConfigData(configToSet)
            });
    }

    /* Queries the whole config object.
       Other getters cherry-pick from this object for utility.
     */
    getConfigData(): Promise<ConfigData> {
        return fetch(`${config.DOMAIN}/config`)
            .then((value: Response) => value.json());
    }

    setConfigData(configData : ConfigData) {

        const headers = new Headers();
        headers.append("x-refresh", sessionStorage.getItem('refreshToken'));
        headers.append("Authorization", `Bearer ${sessionStorage.getItem("accessToken")}`);
        headers.append("Content-Type", "application/json");

        const requestOptions = {
            method: 'PUT',
            headers: headers,
            body: JSON.stringify(configData),
        };

        return new Promise<void>((resolve, reject) => {
            fetch(`${config.DOMAIN}/config`, requestOptions)
                .then(response => {
                    const new_accessToken = response.headers.get('x-access-token');
                    if (new_accessToken) {
                        //if a new accessToken is provided, update it.
                        sessionStorage.setItem("accessToken", new_accessToken);
                    }

                    if (response.status === 200) {
                        resolve();
                    } else {
                        reject();
                    }
                })
                .catch(() => reject());
        })
    }
}