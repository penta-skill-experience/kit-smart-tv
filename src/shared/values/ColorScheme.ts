import {Background} from "./Background";

export interface ColorScheme {
    id: string;
    name: string;
    titleFontColor: string;
    bodyFontColor: string;
    specialBoldFontColor: string;
    specialSubtleFontColor: string;
    accentBarColor: string;
    backgroundImages: Background[];
}