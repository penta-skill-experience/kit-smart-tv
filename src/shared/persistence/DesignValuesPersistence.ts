import {ColorScheme} from "../values/ColorScheme";
import {Background} from "../values/Background";
import {FontSize} from "../values/FontSize";
import config from "./persistence.config.json";
import {ValuesData} from "./data";

export class DesignValuesPersistence {

    getColorSchemeTypes(): Promise<string[]> {
        //todo: this is just a mock
        return new Promise<string[]>(resolve => {
            const data = ["dark", "light"];
            resolve(data);
        });
    }

    getColorScheme(id: string): Promise<ColorScheme> {

        return this.getValuesData().then(values => values.colorSchemes.find(colorScheme => colorScheme.id === id));

        //todo: this is just a mock
        // return new Promise<ColorScheme>(resolve => {
        //     const mockDatabaseColorSchemes: { [key: string]: ColorScheme } = {
        //         "dark": {
        //             id: "dark",
        //             name: "Dark",
        //             titleFontColor: "limeGreen",
        //             bodyFontColor: "white",
        //             specialBoldFontColor: "",
        //             specialSubtleFontColor: "",
        //             accentBarColor: "rgba(48, 48, 48, .45)",
        //             backgroundImages: [new Background("image", "https://images.wallpaperscraft.com/image/single/city_skyscrapers_clouds_rain_road_cars_lights_58563_3840x2160.jpg")]
        //         },
        //         "light": {
        //             id: "light",
        //             name: "Light",
        //             titleFontColor: "royalBlue",
        //             bodyFontColor: "black",
        //             specialBoldFontColor: "red",
        //             specialSubtleFontColor: "orange",
        //             accentBarColor: "rgba(240, 240, 240, .6)",
        //             backgroundImages: [new Background("image", "https://images.wallpapersden.com/image/download/new-york-city-buildings-at-day-sunlight_am5samuUmZqaraWkpJRobWllrWdma2U.jpg")]
        //         },
        //     };
        //     const data = mockDatabaseColorSchemes[id];
        //     resolve(data);
        // });
    }

    setColorSchemes(list: ColorScheme[]) {
        //todo
    }

    getBackground(id: string): Background {
        //todo: do we even need this?

        //todo: this is just a mock
        return {
            id: id,
            url: "someimage.png"
        };
    }

    getBackgrounds(colorSchemeId: string): string[] {
        //todo
        return [];
    }

    setBackgrounds(list: string[]) {
        //todo
    }

    getFontSizes(): Promise<string[]> {
        //todo: this is just a mock
        return new Promise<string[]>(resolve => resolve(["small", "medium", "large"]));
    }

    //todo: do we need this method?
    setFontSizes(list: FontSize[]) {
        //todo
    }

    getFontSize(id: string): Promise<FontSize> {
        return this.getValuesData().then(values => values.fontSizes.find(fontSize => fontSize.id === id));

        //todo: this is just a mock
        // return new Promise<FontSize>(resolve => {
        //     const mockDatabaseColorSchemes = {
        //         "small": {id: "small", name: "Small", relativeSize: 0.8},
        //         "medium": {id: "medium", name: "Medium", relativeSize: 1},
        //         "large": {id: "large", name: "Large", relativeSize: 1.2},
        //     };
        //     const data = mockDatabaseColorSchemes[id];
        //     resolve(data);
        // });
    }

    /* Queries the whole values object.
       Other getters cherry-pick from this object for utility.
     */
    getValuesData(): Promise<ValuesData> {
        return fetch(`${config.DOMAIN}/values`)
            .then((value: Response) => value.json());
    }
}