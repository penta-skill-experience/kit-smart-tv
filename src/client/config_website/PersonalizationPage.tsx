import * as React from "react";
import {FormLabel, Grid} from "@mui/material";
import Radio from '@mui/material/Radio';
import RadioGroup from '@mui/material/RadioGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import FormControl from '@mui/material/FormControl';
import ImageList from '@mui/material/ImageList';
import ImageListItem from '@mui/material/ImageListItem';
import Button from "@mui/material/Button";
import {DesignValuesPersistence} from "../../shared/persistence/DesignValuesPersistence";
import {ColorScheme} from "../../shared/values/ColorScheme";
import {ConfigData, ValuesData} from "../../shared/interfaces/interfaces";
import {DesignConfigPersistence} from "../../shared/persistence/DesignConfigPersistence";

interface PersonalizationPageProps {
    colorScheme: string;
    fontSize: string;
    handleColorSchemeChange: any;
    handleFontSizeChange: any;
    selectedBackground: any;
    handleBackgroundSelect: any;
    handlePersonalizationChange: any;
    children: any;
}

interface PersonalizationPageState {
    loadedDesignState: boolean;
    designValues: ValuesData;
    designConfig: ConfigData;
}

export class PersonalizationPage extends React.Component<PersonalizationPageProps, PersonalizationPageState> {

    constructor(props: Readonly<PersonalizationPageProps> | PersonalizationPageProps) {
        super(props);
        this.state = {
            loadedDesignState: false,
            designValues: undefined,
            designConfig: undefined,
        };
    }

    componentDidMount() {
        this.queryDesignState();
    }

    private queryDesignState(): void {
        new DesignValuesPersistence().getValuesData().then(valuesData => {
            new DesignConfigPersistence().getConfigData().then(configData => this.setState({
                loadedDesignState: true,
                designValues: valuesData,
                designConfig: configData,
            }));
        });
    }

    render() {
        return (
            <div>
                <Grid container spacing={2} direction="row" justifyContent="center" alignItems="center">
                    <Grid item xs={12}>
                        <FormControl>
                            <FormLabel>Choose your color scheme:</FormLabel>
                            <RadioGroup onChange={this.props.handleColorSchemeChange} value={this.props.colorScheme}>
                                {this.state.loadedDesignState &&
                                    this.state.designValues.colorSchemes.map(colorScheme =>
                                        <FormControlLabel
                                            value={colorScheme.id}
                                            control={<Radio/>}
                                            label={colorScheme.name}
                                        />
                                    )}
                            </RadioGroup>
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        <FormControl>
                            <FormLabel>Choose your preferred font size:</FormLabel>
                            <RadioGroup onChange={this.props.handleFontSizeChange} value={this.props.fontSize}>
                                {this.state.loadedDesignState &&
                                    this.state.designValues.fontSizes.map(fontSize =>
                                        <FormControlLabel
                                            value={fontSize.id}
                                            control={<Radio/>}
                                            label={fontSize.name}
                                        />
                                    )}
                            </RadioGroup>
                        </FormControl>
                    </Grid>
                    <Grid item xs={12}>
                        {this.state.loadedDesignState &&
                            this.renderBackgroundImages()}
                    </Grid>
                    <Grid item xs={12}>
                        <Button
                            onClick={this.props.handlePersonalizationChange}
                            variant="outlined"
                        >
                            Save Changes</Button>
                    </Grid>
                </Grid>
            </div>
        );
    }

    private renderBackgroundImages() {

        // selected color scheme:
        const colorScheme: ColorScheme = this.state.designValues.colorSchemes.find(c => c.id === this.props.colorScheme);

        if (!colorScheme) return <></>;

        return <div>
            <h1>Available backgrounds for color scheme {colorScheme.name}:</h1>
            <ImageList sx={{width: 500, height: 450}} cols={3} rowHeight={164}>
                {colorScheme.backgrounds.map(backgroundUrl => (
                    <div>
                        <Grid container spacing={2} direction="column" justifyContent="center"
                              alignItems="center">
                            <Grid item>
                                <ImageListItem key={backgroundUrl}>
                                    <img
                                        src={`${backgroundUrl}?w=164&h=164&fit=crop&auto=format`}
                                        srcSet={`${backgroundUrl}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
                                        loading="lazy"
                                    />
                                </ImageListItem>
                            </Grid>
                            <Grid item>
                                <Radio
                                    checked={this.props.selectedBackground === backgroundUrl}
                                    onChange={event => this.props.handleBackgroundSelect(event.target.value)}
                                    value={backgroundUrl}
                                />
                            </Grid>
                        </Grid>
                    </div>
                ))}
            </ImageList>
        </div>;
    }
}