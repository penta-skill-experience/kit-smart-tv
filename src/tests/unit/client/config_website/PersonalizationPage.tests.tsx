import {shallow, configure} from "enzyme";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import * as React from "react";
import toJson from "enzyme-to-json";
import {PersonalizationPage} from "../../../../client/config_website/PersonalizationPage";
import {DesignValuesPersistence} from "../../../../shared/persistence/DesignValuesPersistence";
import {ConfigData, ValuesData} from "../../../../shared/interfaces/interfaces";
import testImage  from "./test.jpg";
import {FontSize} from "../../../../shared/values/FontSize";
import {ColorScheme} from "../../../../shared/values/ColorScheme";
import {DesignConfigPersistence} from "../../../../shared/persistence/DesignConfigPersistence";

configure({adapter: new Adapter()});

const componentDidMountMock = jest.spyOn(DesignValuesPersistence.prototype,"getValuesData");
const setConfigDataMock = jest.spyOn(DesignConfigPersistence.prototype, "setConfigData");


const testConfigData = {
    fontSize: "small",
    colorScheme: "light",
    background: "test.img",
}

let fontSizeList : FontSize[] = [{id: "1", name: "arial", relativeSize: 5}];
let colorSchemeList : ColorScheme[] = [{
    id: "1",
    name: "light",
    titleFontColor: "blue",
    bodyFontColor: "yellow",
    specialBoldFontColor: "black",
    specialSubtleFontColor: "green",
    accentBarColor: "pink",
    backgrounds: [testImage],

}];

const testValuesData = {
    fontSizes: fontSizeList,
    colorSchemes: colorSchemeList,
} as ValuesData;

componentDidMountMock.mockImplementation(() => {
    return new Promise<ValuesData>(resolve => {
    });
});

setConfigDataMock.mockImplementation(() => {
    return new Promise<void>(resolve => {});
});

let wrapper;
const method = () => {
};

afterAll(() => {
    jest.restoreAllMocks();
})

describe("Personalization page tests", () => {

    beforeEach(() => {
        wrapper = shallow(
            <PersonalizationPage
                colorScheme={''}
                fontSize={''}
                handleColorSchemeChange={method}
                handleFontSizeChange={method}
                selectedBackground={method}
                handleBackgroundSelect={method}
                handlePersonalizationChange={method}
            >
            </PersonalizationPage>
        );
    });


    test("snapshot test for personalization page", async () => {
        wrapper.setState({
            loadedDesignState: true,
            designValues: testValuesData,
            designConfig: testConfigData,
            successfulBar:false,
            errorBar:false,
            sessionErrorBar:false,
        });
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test("snapshot for render background function", async () => {
        const backgroundWrapper = shallow(
            <PersonalizationPage
                colorScheme={'1'}
                fontSize={''}
                handleColorSchemeChange={method}
                handleFontSizeChange={method}
                selectedBackground={method}
                handleBackgroundSelect={method}
                handlePersonalizationChange={method}
            >
            </PersonalizationPage>
        )
        expect(toJson(backgroundWrapper)).toMatchSnapshot();
    });

    test("handleClose function test", () => {
        wrapper.find("ForwardRef(Snackbar)").at(0).simulate("close");
        expect(wrapper.find("ForwardRef(Snackbar)").at(0).prop("open")).toEqual(false);
    });

    test("handleClose function if case test", () => {
        wrapper.find("ForwardRef(Snackbar)").at(0).simulate("close", { target: { event: '', reason: 'clickaway'} });
        expect(wrapper.find("ForwardRef(Snackbar)").at(0).prop("open")).toEqual(false);
    });
});

describe("snack bar test", () => {

    test("handlePersonalization change test", () => {
        const handlePersonalizationChangeThenMock = jest.fn(() => Promise.resolve(true));
        wrapper = shallow(
            <PersonalizationPage
                colorScheme={''}
                fontSize={''}
                handleColorSchemeChange={method}
                handleFontSizeChange={method}
                selectedBackground={method}
                handleBackgroundSelect={method}
                handlePersonalizationChange={handlePersonalizationChangeThenMock}
            >
            </PersonalizationPage>
        );

        wrapper.find("ForwardRef(Button)").at(0).simulate("click");
        expect(handlePersonalizationChangeThenMock).toHaveBeenCalled();
    });

    test("handlePersonalization change test", () => {
        const handlePersonalizationChangeCatchMock = jest.fn(() => Promise.resolve(false));
        wrapper = shallow(
            <PersonalizationPage
                colorScheme={''}
                fontSize={''}
                handleColorSchemeChange={method}
                handleFontSizeChange={method}
                selectedBackground={method}
                handleBackgroundSelect={method}
                handlePersonalizationChange={handlePersonalizationChangeCatchMock}
            >
            </PersonalizationPage>
        );

        wrapper.find("ForwardRef(Button)").at(0).simulate("click");
        expect(handlePersonalizationChangeCatchMock).toHaveBeenCalled();
    });

});

describe("background image render test", () => {

    test("snapshot for render background function", async () => {
        const backgroundWrapper = shallow(
            <PersonalizationPage
                colorScheme={'1'}
                fontSize={''}
                handleColorSchemeChange={method}
                handleFontSizeChange={method}
                selectedBackground={method}
                handleBackgroundSelect={method}
                handlePersonalizationChange={method}
            >
            </PersonalizationPage>
        )
        expect(toJson(backgroundWrapper)).toMatchSnapshot();
    });
});