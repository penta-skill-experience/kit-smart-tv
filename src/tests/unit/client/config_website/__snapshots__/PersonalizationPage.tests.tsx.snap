// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Personalization page tests snapshot for render background function 1`] = `
<div>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    direction="row"
    justifyContent="center"
    spacing={2}
  >
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your color scheme:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value="1"
        />
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your preferred font size:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value=""
        />
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    />
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        Save Changes
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Changes Saved"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Color scheme, font size and a background must be chosen"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Session expired"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;

exports[`Personalization page tests snapshot test for personalization page 1`] = `
<div>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    direction="row"
    justifyContent="center"
    spacing={2}
  >
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your color scheme:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value=""
        >
          <ForwardRef(FormControlLabel)
            control={<ForwardRef(Radio) />}
            label="light"
            value="1"
          />
        </ForwardRef(RadioGroup)>
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your preferred font size:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value=""
        >
          <ForwardRef(FormControlLabel)
            control={<ForwardRef(Radio) />}
            label="arial"
            value="1"
          />
        </ForwardRef(RadioGroup)>
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    />
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        Save Changes
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Changes Saved"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Color scheme, font size and a background must be chosen"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Session expired"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;

exports[`background image render test snapshot for render background function 1`] = `
<div>
  <ForwardRef(Grid)
    alignItems="center"
    container={true}
    direction="row"
    justifyContent="center"
    spacing={2}
  >
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your color scheme:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value="1"
        />
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(FormControl)>
        <ForwardRef(FormLabel)>
          Choose your preferred font size:
        </ForwardRef(FormLabel)>
        <ForwardRef(RadioGroup)
          onChange={[Function]}
          value=""
        />
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
    <ForwardRef(Grid)
      item={true}
      xs={12}
    />
    <ForwardRef(Grid)
      item={true}
      xs={12}
    >
      <ForwardRef(Button)
        onClick={[Function]}
        variant="contained"
      >
        Save Changes
      </ForwardRef(Button)>
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Changes Saved"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Color scheme, font size and a background must be chosen"
        onClose={[Function]}
        open={false}
      />
      <ForwardRef(Snackbar)
        anchorOrigin={
          Object {
            "horizontal": "right",
            "vertical": "bottom",
          }
        }
        autoHideDuration={2000}
        message="Session expired"
        onClose={[Function]}
        open={false}
      />
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;
